adobe/spectrum-web-components
# fix(overlay): normalize popover toggling between native and synthetic [popover] usage

## Description
Make sure that [this demo](https://cleanup--spectrum-web-components.netlify.app/storybook/?path=/story/overlay-element--click) performs the same way regardless of whether the Overlay API is actively leveraging `[popover]` or polyfilling it.

Also includes:
- bump lowest Node version
- correct `debounce` usage due to dependabot version bump

## Related issue(s)
- fixes #3641

## How has this been tested?
    1. Go [here](https://cleanup--spectrum-web-components.netlify.app/storybook/?path=/story/overlay-element--click)
    2. Visit in Chrome (uses `[popover]`)
    3. Click the button over and over again
    4. See that the Overlay is toggled
    1. Go [here](https://cleanup--spectrum-web-components.netlify.app/storybook/?path=/story/overlay-element--click)
    2. Visit in Firefox (does not use `[popover]`)
    3. Click the button over and over again
    4. See that the Overlay is toggled

## Types of changes

## Checklist
